runner/execer/os/os_execer.go

```diff
@@ -395,29 +388,141 @@ func (p *osProcess) Wait() (result execer.ProcessStatus) {
 	return result
 }
 
-func (p *osProcess) Abort() (result execer.ProcessStatus) {
+// Attempt to SIGTERM process, allowing for graceful exit
+// SIGKILL after 10 seconds or if osProcess.cmd.Wait() returns an error
+func (p *osProcess) Abort() execer.ProcessStatus {
 	p.mutex.Lock()
 	defer p.mutex.Unlock()
 	if p.result != nil {
 		return *p.result
 	} else {
-		p.result = &result
+		p.result = &execer.ProcessStatus{}
 	}
-	result.State = execer.FAILED
-	result.ExitCode = -1
-	result.Error = "Aborted."
+	p.result.State = execer.FAILED
+	p.result.ExitCode = -1
+	p.result.Error = "Aborted."
+
+	sigCh := make(chan error)
+	var err error
+
+	go func() {
+		sigCh <- p.cmd.Process.Signal(syscall.SIGTERM)
```

Comment:
why async just for this function? we can just call and check err immediately

Reply:
ah good catch - relic from previous iteration of this PR i forgot to change.